internal/audit/hub/hub.go

```diff
@@ -252,29 +255,45 @@ func (l *Log) streamPrefix(ctx context.Context, kind logsv1.IngestBatch_EntryKin
 
 	stream.Send = func(buf *z.Buffer) error {
 		logger.Log(zapcore.Level(-3), "Converting buffer to key-values")
-		kvList, err := badgerv4.BufferToKVList(buf)
-		if err != nil {
-			return fmt.Errorf("failed to convert buffer to key-values: %w", err)
-		}
 
-		keys := make([][]byte, len(kvList.Kv))
-		for i, kv := range kvList.Kv {
-			keys[i] = kv.Key
+		syncKeys := func(keys [][]byte) error {
+			logger.Log(zapcore.Level(-3), "Syncing and deleting batch")
+			if err := l.syncThenDelete(ctx, kind, keys); err != nil {
+				return fmt.Errorf("failed to sync and delete logs: %w", err)
+			}
+
+			return nil
 		}
 
-		for i := 0; i < len(keys); i += l.maxBatchSize {
-			end := i + l.maxBatchSize
-			if end > len(keys) {
-				end = len(keys)
+		kv := kvPool.Get().(*bpb.KV) //nolint:forcetypeassert
+		defer kvPool.Put(kv)
+
+		var i int
+		keys := make([][]byte, l.maxBatchSize)
+		if err := buf.SliceIterate(func(s []byte) error {
+			kv.Reset()
+
+			if err := kv.Unmarshal(s); err != nil {
+				return err
 			}
 
-			logger.Log(zapcore.Level(-3), "Syncing and deleting batch")
-			if err := l.syncThenDelete(ctx, kind, keys[i:end]); err != nil {
-				return fmt.Errorf("failed to sync and delete logs: %w", err)
+			keys[i] = append([]byte(nil), kv.Key...)
```

Comment:
Aren't we losing a little bit of the pool advantage by allocating a new byte slice here for each key? I think we have two options here.
- Take a new KV from the pool for each item until we reach the batch size limit, sync those and then release all of them back to the pool
- Have a second pool holding `[l.maxBatchSize][]byte` objects. [`clear`](https://pkg.go.dev/builtin#clear) then  [`copy`](https://pkg.go.dev/builtin#copy)  the key into each entry. (We might not even need the `clear` because keys are fixed size but it's probably better to be safe.)